ark/internal/genai/team.go

```diff
@@ -86,33 +86,56 @@ func (t *Team) executeRoundRobin(ctx context.Context, userInput Message, history
 	messages := slices.Clone(history)
 	var newMessages []Message
 
-	for turn := 0; ; turn++ {
+	messageCount := 0 // Count individual agent messages
+	memberIndex := 0  // Track which agent should speak next
+
+	for {
 		// Check if context was cancelled
 		if ctx.Err() != nil {
 			return newMessages, ctx.Err()
 		}
 
-		turnTracker := NewExecutionRecorder(t.Recorder)
-		turnTracker.TeamTurn(ctx, "Start", t.FullName(), t.Strategy, turn)
+		// Check maxTurns before executing
+		if t.MaxTurns != nil && messageCount >= *t.MaxTurns {
+			turnTracker := NewExecutionRecorder(t.Recorder)
+			turnTracker.TeamTurn(ctx, "MaxTurns", t.FullName(), t.Strategy, messageCount)
+			// Log the maxTurns limit for observability, but return success with accumulated messages
+			// This preserves monitoring/alerting capabilities
+			t.Recorder.EmitEvent(ctx, "Warning", "TeamMaxTurnsReached", BaseEvent{
```

Comment:
nice - there's a constant we can use for `Warning` if you check the docs for emit event, its in ./docs somewhere